### PR TITLE
Add linkerd dashboard and monitoring configuration

### PR DESCRIPTION
**Pull request by phillebaba**

### CODE REVIEW

File: modules/kubernetes/linkerd/main.tf [hunk 1]
```diff
@@ -216,3 +232,17 @@ resource "helm_release" "linkerd" {
     }),
   ]
 }
+
+# Install linkerd viz extension https://github.com/linkerd/linkerd2/tree/main/viz/charts/linkerd-viz
+resource "helm_release" "linkerd_viz" {
```
[the remark this answers is not in the document]
Comment (phillebaba) [reply modules/kubernetes/linkerd/main.tf:237]: Nope nothing is needed.

File: modules/kubernetes/linkerd/main.tf [hunk 2]
```diff
@@ -100,6 +100,22 @@ resource "kubernetes_namespace" "cni" {
...
+resource "kubernetes_namespace" "viz" {
+  metadata {
+    labels = {
+      name                   = "linkerd-viz"
+      "xkf.xenit.io/kind"    = "platform"
+      "linkerd.io/extension" = "viz"
+    }
+    annotations = {
+      #"viz.linkerd.io/external-prometheus" = "{{.Values.prometheusUrl}}"
+      "linkerd.io/inject"             = "enabled"
```
[the remark this answers is not in the document]
Comment (phillebaba) [reply modules/kubernetes/linkerd/main.tf:112]: All namespace labels and annotations are taken from the helm chart.

File: modules/kubernetes/prometheus/charts/prometheus-extras/templates/monitor.yaml [hunk 3]
```diff
@@ -184,3 +184,36 @@ spec:
...
       app.kubernetes.io/name: falco-exporter
+{{- if .Values.enabledMonitors.linkerd }}
+---
+apiVersion: monitoring.coreos.com/v1
+kind: ServiceMonitor
+metadata:
+  name: linkerd-federate
+  namespace: linkerd-viz
+  labels:
+    xkf.xenit.io/monitoring: platform
```
[the remark this answers is not in the document]
Comment (phillebaba) [reply modules/kubernetes/prometheus/charts/prometheus-extras/templates/monitor.yaml:195]: Dont see the need right now.

File: modules/kubernetes/linkerd/main.tf [hunk 4]
```diff
@@ -100,6 +100,22 @@ resource "kubernetes_namespace" "cni" {
   }
 }
 
+resource "kubernetes_namespace" "viz" {
+  metadata {
+    labels = {
+      name                   = "linkerd-viz"
+      "xkf.xenit.io/kind"    = "platform"
```
[the remark this answers is not in the document]
Comment (phillebaba) [reply modules/kubernetes/linkerd/main.tf:107]: No

**Review by simongottschlag — APPROVED** [review]
LGTM! :1st_place_medal:

**Pull request merged**